### PR TITLE
feature(parallized deploy): allow configuration of the deployer to run tasks in parallel (per leaf node)

### PR DESCRIPTION
**Pull request by dakaneye**
**Summary of Changes**
- Configuration: Adds a deployThreads configuration.
- MavenRepositoryDeployer: establishes this class as a Singleton
- MavenRepositoryDeploymentCallable: for a given leaf node, this does the deployment, returning a new DeploymentResult object to collect for the overall summary

### CODE REVIEW

**Comment by dakaneye** [comment]
@mosabua FYI! I couldn't get the checkstyle to pass locally, would you be interested in porting over to spotless?

**Comment by mosabua** [comment]
Please change commit messages to follow https://trino.io/development/process#pull-request-and-commit-guidelines-

File: maven-repository-provisioner/src/test/java/com/simpligility/maven/provisioner/MavenRepositoryDeploymentCallableTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.Assert.assertEquals;
+
+public class MavenRepositoryDeploymentCallableTest
```
Comment (mosabua) [inline maven-repository-provisioner/src/test/java/com/simpligility/maven/provisioner/MavenRepositoryDeploymentCallableTest.java:19]: awesome .. definitely need more tests overall ;-)

File: maven-repository-provisioner/src/test/java/com/simpligility/maven/provisioner/helpers/RepositorySystemImpl.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
+package com.simpligility.maven.provisioner.helpers;
+
+import org.eclipse.aether.RepositorySystem;
```
Comment (mosabua) [inline maven-repository-provisioner/src/test/java/com/simpligility/maven/provisioner/helpers/RepositorySystemImpl.java:3]: We should really move to the newer version of aether or the maven resolver replacement .. but that can follow later cc @cstamas

**Review by mosabua — CHANGES_REQUESTED** [review]
Had a first look .. once its compiles I can do some testing ..

File: .gitignore [hunk 3]
```diff
@@ -10,3 +10,4 @@ release.properties
 local-cache
 maven-repository-provisioner.log
 .checkstyle
+**/*.iml
```
[the remark this answers is not in the document]
Comment (dakaneye) [reply .gitignore:13]: i'm not 100% sure how it ended up in git tracking, but I had one generated when I tried to add some checkstyle plugins to my intellij.

**Comment by cstamas** [comment]
Folks, unsure what you want to achieve, but parallel deploy in Resolver 1.9+ is a thing. All you need is to "batch" the deploy request. See https://issues.apache.org/jira/browse/MRESOLVER-319

**Comment by mosabua** [comment]
> Folks, unsure what you want to achieve, but parallel deploy in Resolver 1.9+ is a thing. All you need is to "batch" the deploy request. See https://issues.apache.org/jira/browse/MRESOLVER-319


I vaguely remember now .. but yes.. for that to be used we have to upgrade this project to the new resolver stuff .. which arguably would be good.

**Pull request closed without merging**